debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelopeTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.connector.mongodb.transforms;
+
+import static org.fest.assertions.Assertions.assertThat;
+import java.util.Collections;
+import java.util.Map.Entry;
+
+import org.apache.kafka.connect.data.Schema;
+import org.apache.kafka.connect.data.SchemaBuilder;
+import org.apache.kafka.connect.data.Struct;
+import org.apache.kafka.connect.sink.SinkRecord;
+import org.apache.kafka.connect.transforms.ExtractField;
+import org.bson.BsonDocument;
+import org.bson.BsonValue;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.debezium.data.Envelope.FieldName;
+import io.debezium.data.Json;
+/**
+ * Unit test for {@code StringToJSON}.
+ *
+ * @author Sairam Polavarapu
+ */
+public class UnwrapFromMongoDbEnvelopeTest {
```

Comment:
Tbh. I don't think this test makes too much sense. Most notably, it doesn't invoke the SMT `UnwrapFromMongoDbEnvelope` at all. It seems more like it re-implements it? It's not quite clear how this supposed to work. I'll push an example for how it should look like in a bit.

Reply:
Test is largely re-written now as per my update.